ChangeLog

```diff
@@ -9,6 +9,9 @@ Release date: TBA
 * Add new (optional) ``doc_node`` attribute to ``nodes.Module``, ``nodes.ClassDef``,
   and ``nodes.FunctionDef``.
 
+* Accessing the ``doc`` attribute of ``nodes.Module``, ``nodes.ClassDef``, and
+  ``nodes.FunctionDef`` has been deprecated in favour of the ``doc_node.value`` attribute.
```

Comment:
```suggestion
* Accessing the ``doc`` attribute of ``nodes.Module``, ``nodes.ClassDef``, and
  ``nodes.FunctionDef`` has been deprecated in favour of the ``doc_node`` attribute.
```
Maybe we should add an additional note here that captures the concerns you have raised? Something like this?
```
Note: ``doc_node`` is ``nodes.Const | None`` whereas ``doc`` was ``str | None``.
```

Reply:
```suggestion
* Accessing the ``doc`` attribute of ``nodes.Module``, ``nodes.ClassDef``, and
  ``nodes.FunctionDef`` has been deprecated in favour of the ``doc_node`` attribute.
  Note: ``doc_node`` is ``nodes.Const | None`` whereas ``doc`` was ``str | None``.
```

Reply:
I like the note (committed). Should we also mention `.value` or should people (we and who else...?) figure that out themselves?

Reply:
If you have a good idea, go for it! We just need to be cautious that `doc_node` can be `None` and thus `doc_node.value` won't always work.

Reply:
Let's leave as is. Not really helpful to create a very long changelog entry which is likely only useful to us and current `pylint` contributors.